src/server/dbFactory.js

```diff
@@ -0,0 +1,36 @@
+const PgSqlProvider = require("./Generators/classes/pgSqlProvider");
```

Comment:
Can you provide an article referencing the pattern you used here and maybe comment a little on why it you decided to use this pattern pleaseeee

Reply:
It's loosely based on the factory design pattern https://www.tutorialspoint.com/design_pattern/factory_pattern.htm